### PR TITLE
Add UserPermission to Ability.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Note that only the commit b4dc238 is part of this PR, the earlier PR is part of #8718 on which this builds.

### CODE REVIEW

**Comment by ashercodeorg** [comment]
CC @jeremydstone That this was missing indicates we should do a thorough audit of abilities.

File: dashboard/app/models/ability.rb [hunk 1]
```diff
@@ -167,23 +163,13 @@ def initialize(user)
...
-    #
-    # The third argument is an optional hash of conditions to further filter the
-    # objects.
-    # For example, here the user can only update published articles.
-    #
-    #   can :update, Article, :published => true
-    #
-    # See the wiki for details:
-    # https://github.com/ryanb/cancan/wiki/Defining-Abilities
+    if user.admin
```
Comment (laurelfan) [inline dashboard/app/models/ability.rb]: nit: `user.admin?` would be consistent with above

File: dashboard/app/models/ability.rb [hunk 2]
```diff
@@ -1,53 +1,47 @@
...
+      Pd::Workshop,
+      Pd::Attendance,
+      Pd::DistrictPaymentTerm,
+      Pd::DistrictReport,
+      Pd::WorkshopOrganizerReport,
+      Pd::TeacherProgressReport,
+      Pd::CourseFacilitator
+    ]
 
     if user.id
```
Comment (laurelfan) [inline dashboard/app/models/ability.rb]: while you're refactoring, `user.persisted?` might be a more meaningful way to say this

**Comment by laurelfan** [comment]
LGTM. Note that this ability/inability doesn't actually do anything because there is no controller code that checks it (cancan is magical but not that magical -- it doesn't like prevent sql updates based on the logged in user or anything, you have to call it more or less explicitly -- you would look for an `authorize!` call in a controller that references the UserPermission class or an instance of it, or a `load_and_authorize_resource` before_action in a UserPermissionsController, which does not exist). This would imply that the authorization checking for whatever is modifying UserPermissions is happening outside cancan (because it's in pegasus?).

**Comment by ashercodeorg** [comment]
The modifications to `UserPermission` are indeed in pegasus-land (e.g., https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/sites.v3/code.org/public/private/privileges/index.haml).

But there are existing reads of `UserPermission` in the `User` model.

**Comment by laurelfan** [comment]
yep, the ability stuff does not affect any of those usages of
UserPermission.

**Pull request merged**